lib/matplotlib/artist.py

```diff
@@ -1417,13 +1417,18 @@ def get_aliases(self):
            'linewidth'      : {'lw'},
           }
         """
-        names = [name for name in dir(self.o)
+        return ArtistInspector._get_aliases_class(self.o)
+
+    @lru_cache(maxsize=None)
+    @staticmethod
+    def _get_aliases_class(o):
```

Comment:
```suggestion
    def _get_aliases(class_):
        """Cached implementation of `.get_aliases`."""
```
ArtistInspector is not written very cleary, but `o` is actually a class. So let's call it that. I suggest to use `class_` instead of `cls` because it's the first argument in the method and `method(cls)` is typically used for class methods.